src/main/java/com/jpop/productservice/controller/ProductServiceController.java

```diff
@@ -0,0 +1,67 @@
+package com.jpop.productservice.controller;
+
+import com.google.gson.Gson;
+import com.jpop.productservice.model.Product;
+import com.jpop.productservice.service.ProductDeleteService;
+import com.jpop.productservice.service.ProductDetailService;
+import com.jpop.productservice.service.ProductInsertService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/api/v1/products")
+public class ProductServiceController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProductServiceController.class);
+    private ProductInsertService productInsertService;
+    private ProductDetailService productDetailService;
+    private ProductDeleteService productDeleteService;

```

Comment:
I felt it's an overhead to write different classes for each CRUD operation. Is this style recommended somewhere?

Reply:
Can you please answer this.

Reply:
I have read one article and according to that class should not have more than 150 lines and crud operations can be done in separate class.